### DISCUSSION TITLE
Good way to do tls.cert -> certificate thumb without CLI

### DISCUSSION THREAD

**Discussion by futurepaul** [btcpayserver/btcpayserver#3361] [Ideas / Feature Requests]
Most of a BTCPay server setup can be done without CLI knowledge, however "LND via the REST proxy" has this scary line to get the certthumbprint:

`openssl x509 -noout -fingerprint -sha256 -in /root/.lnd/tls.cert | sed -e 's/.*=//;s/://g'`

One potential fix is to have node distros include this with their connect info ([Raspiblitz does this](https://github.com/rootzoll/raspiblitz/blob/a5c5f455c71f6a7b715f485309216ee5321e2d7f/home.admin/config.scripts/lnd.export.sh#L98))

Is there a better way? Should this be added to LND itself? Can this be added to the BTCPay gui so you can paste in a tls.cert?

**Comment by futurepaul** [comment]
Potentially related to https://github.com/btcpayserver/btcpayserver/issues/2182

**Reply by dennisreimann** [reply]
Can you elaborate on the use case? The command you mentioned is listed in the custom node setup info. In case you want to set up another Lightning node to be used inside a BTCPay store, we couldn't retrieve its certificate fingerprint, because it's on a remote node.

**Reply by futurepaul** [reply]
yeah, the use case is a lightning node runner that wants btcpay but isn't comfortable with the command line. likely they can get their tls cert but they might be scared to / confused about running this command against it

**Reply by beeforbacon** [reply]
A simple way to generate the string BTCPAY needs to interact with an external lnd node would go a long way for merchants to be able to use btpcay.  Imagine spooling up a btcpay and then just dragging in the macaroon/cert file and then btcpay in the background generates the hex string needed to connect. Not every node runner has access to or knows how to use a command line.

**Reply by dennisreimann** [reply]
Yes, but this cannot be done via BTCPay itself, because those files reside on a remote node then. However, I think it'd be good if those node systems (RaspiBlitz, Umbrel et al) or Lightning services could provide the connection string via their UI. 

I got a simple script that assembles everything (here's an example for LND). This could be integrated and customized, depending on the Lightning node (e.g. for c-lightning) and used to retrieve the info that the user wants:

```sh
certPath=tls.cert
macaroonPath=admin.macaroon
thumbprint=$(openssl x509 -noout -fingerprint -sha256 -in $certPath | sed -e 's/.*=//;s/://g')
macaroon=$(xxd -plain $macaroonPath | tr -d '\n')
connection="type=lnd-rest;server=https://$lnHost$lnPort/;macaroon=$macaroon;certthumbprint=$thumbprint"
```
